src/Shared/Shared.csproj

```diff
@@ -26,12 +26,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.Memory" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
-    <PackageReference Include="Microsoft.Extensions.ObjectPool" />
-    <PackageReference Include="System.ComponentModel.Annotations" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))" />
-    <PackageReference Include="System.ValueTuple" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))" />
     <PackageReference Include="Microsoft.Bcl.HashCode" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
+    <PackageReference Include="Microsoft.Extensions.ObjectPool" NoWarn="RT0003" />
```

Comment:
@geeknoid would this be the right fix for
```
##[error]CSC(0,0): error RT0003: (NETCORE_ENGINEERING_TELEMETRY=Build) PackageReference Microsoft.Extensions.ObjectPool can be removed
```
https://dev.azure.com/dnceng-public/public/_build/results?buildId=610707&view=logs&j=774c6b69-2ab9-5dfa-262f-030e439e3fb3&t=7ef8a141-91c2-5bcb-07e0-8fdb3656e5b0

Reply:
Lol! @geeknoid and I were just in a call discussing this. We are still taking a look as it seems like this may be a bug in ReferenceTrimmer tool

Reply:
I don't think so.

Shared.csproj is only used so that we can run unit tests on the shared code. In the normal source base, we use <Compile> to pull in sources from the src/Shared folder into other components and Shared.csproj is not involved.

Reply:
Ok, I have a fix for this, will push in a sec.

Reply:
This does the trick without having to add NoWarn as it actually addresses the issue: f9aada2739b44a3b1cee151731595fd6fe5ff8fb